webapp/tests/karma/unit/services/user-settings.js

```diff
@@ -92,8 +92,13 @@ describe('UserSettings service', () => {
 
     const firstPromise = service();
     service();
-    service().then(isExpected);
-    firstPromise.then(isExpected);
+    service()
+      .then(isExpected)
+      .catch(err => done(err));
+    firstPromise
+      .then(isExpected)
+      .catch(err => done(err));
+    done();
```

Comment:
You can't call `done` here because it will execute before the async expectations above have executed. Consider chaining them together, eg:

```suggestion
    return service()
      .then(isExpected)
      .then(() => firstPromise)
      .then(isExpected);
```